networkmanager/src/main/java/com/dezen/riccardo/networkmanager/NetworkDictionary.java

```diff
@@ -0,0 +1,318 @@
+package com.dezen.riccardo.networkmanager;
+
+import android.content.Context;
+
+import androidx.room.Room;
+
+import com.dezen.riccardo.networkmanager.database_dictionary.PeerDatabase;
+import com.dezen.riccardo.networkmanager.database_dictionary.PeerEntity;
+import com.dezen.riccardo.networkmanager.database_dictionary.ResourceDatabase;
+import com.dezen.riccardo.networkmanager.database_dictionary.ResourceEntity;
+import com.dezen.riccardo.smshandler.SMSPeer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.dezen.riccardo.smshandler.SmsHandler.SMS_HANDLER_LOCAL_DATABASE;
+
+/**
+ * Class implementing Dictionary. Conceived as a double dictionary on SMSPeer and StringResource.
+ * Due to trouble with testing android class. Maps have been used.
+ * @author Riccardo De Zen, Giorgia Bortoletti
+ */
+public class NetworkDictionary implements Dictionary<SMSPeer, StringResource> {
+    private Map<String, String>  peers;
+    private Map<String, String> resources;
+    private NetworkDictionaryDatabase database;
+
+    public NetworkDictionary(Context context){
+        database = new NetworkDictionaryDatabase(context);
+        peers = new HashMap<>();
+        resources = new HashMap<>();
+    }
+
+    //TODO? make this method asynchronous
+    /**
+     * Imports resources and peers from database to variables Map
+     */
+    private void importFromDatabase(){
```

Comment:
Should return something (a boolean or throw an exception?) to know if something didn't work

Reply:
Ok thanks, I'll improve it

Reply:
Sorry for barging in.
I don't know if you were intended to follow Barca's suggestion... If that's the case, I think you just forgot to edit this method... I see it's still void... Am I seeing an outdated version?
P.S. I wrote this just to let you know, if you really forgot this ;)